### PR TITLE
fix: drop internal tables when drop table/index

### PR DESCRIPTION
**Pull request by yezizp2012**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

As title. Currently `drop_table` and `drop_index` will only drop the mv, the related internal tables are left behind. They should be dropped along with the mv.

Please explain **IN DETAIL** what the changes are in this PR and why they are needed:

- Summarize your change (**mandatory**)
- How does this PR work? Need a brief introduction for the changed logic (optional)
- Describe clearly one logical change and avoid lazy messages (optional)
- Describe any limitations of the current code (optional)

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

If your pull request contains user-facing changes, please specify the types of the changes, and create a release note. Otherwise, please feel free to remove this section.

### Types of user-facing changes

Please keep the types that apply to your changes, and remove those that do not apply.

* Installation and deployment 
* Connector (sources & sinks)
* SQL commands, functions, and operators
* RisingWave cluster configuration changes
* Other (please specify in the release note below)

### Release note

Please create a release note for your changes. In the release note, focus on the impact on users, and mention the environment or conditions where the impact may occur.

### CODE REVIEW

**Review by HuaHuaY — APPROVED** [review]
LGTM!

File: src/meta/src/manager/catalog.rs [hunk 1]
```diff
@@ -335,7 +334,11 @@ where
         }
     }
 
-    pub async fn drop_table(&self, table_id: TableId) -> MetaResult<NotificationVersion> {
+    pub async fn drop_table(
+        &self,
+        table_id: TableId,
+        internal_table_ids: Vec<TableId>,
```
Comment (BugenZhao) [inline src/meta/src/manager/catalog.rs:340]: I think there should be no internal tables for `TableV2`?

Comment (yezizp2012) [reply src/meta/src/manager/catalog.rs:340]: Yes, it has. 😂 Table will have several internal tables if it got internal states. These will be stored as 'Table' in catalog manager and synced to frontend. 
Note that: calling 'show materialized views' will filter out internal states.

Comment (BugenZhao) [reply src/meta/src/manager/catalog.rs:340]: Do you mean that the table here can also ref to a materialized view?

Comment (yezizp2012) [reply src/meta/src/manager/catalog.rs:340]: Yep, table here is actually a materialized view.

**Pull request merged**